### PR TITLE
Allow integer defaults for double fields

### LINKED ISSUE #150: Doubles with integer values break parser
I think we might have gotten guard-happy
This breaks the parser.
```thrift
struct ImBadAtInitializing {
   1: double myBad = 0;
}
```

### PR DESCRIPTION
**Pull request by pguillory**
Fixes #150.

### CODE REVIEW

File: test/generator/binary_protocol_test.exs [hunk 1]
```diff
@@ -444,4 +444,45 @@ defmodule Thrift.Generator.BinaryProtocolTest do
...
+    assert struct.byte_val == 5
+    assert struct.double_from_float == 0
+    assert is_float(struct.double_from_float)
+    assert struct.double_from_int == 0
+    assert is_integer(struct.double_from_int)
+    assert struct.i16_val == 5
+    assert struct.i32_val == 5
+    assert struct.i64_val == 5
+    assert struct.string_val == "abc123"
+    # TODO: struct.struct_val
```
Comment (pinterest-ebert) [inline test/generator/binary_protocol_test.exs]: Found a TODO tag in a comment: # TODO: struct.struct_val

**Comment by pinterest-ebert** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/pinterest/elixir-thrift/pulls/151.

File: test/generator/binary_protocol_test.exs [hunk 2]
```diff
@@ -444,4 +444,45 @@ defmodule Thrift.Generator.BinaryProtocolTest do
...
+  @thrift_file name: "defaults.thrift", contents: """
+  struct Defaults {
+    1: bool bool_from_true = true
+    2: bool bool_from_false = false
+    3: bool bool_from_one = 1
+    4: bool bool_from_zero = 0
+    5: byte byte_val = 5
+    6: double double_from_float = 0.0
+    7: double double_from_int = 0
+    8: i16 i16_val = 5
```
Comment (scohen) [inline test/generator/binary_protocol_test.exs]: Do we  allow the other way? like
```thrift
 9: i64 i64_from_float = 64.0
```
?

Comment (pguillory) [inline test/generator/binary_protocol_test.exs]: I don't think it allows it currently.

**Pull request merged**